### ISSUE TITLE
Main Page Front-End Implementation

### ISSUE THREAD

**Issue by asyasusen** [bounswe/bounswe2024group12#132]
## Description

We need to implement the game page (that you land on when you login) according to mock-ups. For now using a few generic templates is okay until the back-end is complete.
***

## Acceptance Criteria
- [x] Menu
- [x] Game Of The Day
- [x] Lists (Popular games, new games, more games)
- [x] Reviews (Recent Review, Friend's reviews)
- [x] Implement backend functions
<br>

### Reviewer:
This issue will be reviewed by: @sonerkuyar @ahmetfirat23 @yusufaygun

**Comment by asyasusen** [comment]
@ahmetfirat23  should we use a star rating system or number format?

**Comment by ahmetfirat23** [comment]
Probably a numeric format would give more precise information. We should have mention that in the requirements too I guess #123

**Comment by isilsukarakuzuu** [comment]
@asyasusen how about naming main-page branch front-end/main-page or something like this to be more clear, since later we can be confused by names in backend-frontend?

**Comment by ahmetfirat23** [comment]
I took a look at the main page. It looks nice but have a few feedbacks:

Following requirement 1.2.3.2. our main page includes not only game of the day but property of the day. I realize these weren't in the mockups but this is one of the main elements of our website. It should be right after game of the day and say things like Genre of the Day, Developer of the Day etc. I think three property:value pair is enough for this. These properties will be randomly selected from a premade list of properties and values will be randomly assigned by backend. After completing this please tag me again for reviewing I will close this issue and open a new one for the following ones.

Followings are also requirements but not as much urgent. As requirements 1.2.3.3. says we showcase a few games that were launched on that day. Probably backend will not be able to prioritize this for the milestone but it would be nice if have the section for it.

Lastly, I think showing popular and new games in a single row would look nicer. Four games for each category would be enough for this. Maybe we can implement swiping if we have time. What do you think?

**Comment by asyasusen** [comment]
> I took a look at the main page. It looks nice but have a few feedbacks:
> 
> Following requirement 1.2.3.2. our main page includes not only game of the day but property of the day. I realize these weren't in the mockups but this is one of the main elements of our website. It should be right after game of the day and say things like Genre of the Day, Developer of the Day etc. I think three property:value pair is enough for this. These properties will be randomly selected from a premade list of properties and values will be randomly assigned by backend. After completing this please tag me again for reviewing I will close this issue and open a new one for the following ones.
> 
> Followings are also requirements but not as much urgent. As requirements 1.2.3.3. says we showcase a few games that were launched on that day. Probably backend will not be able to prioritize this for the milestone but it would be nice if have the section for it.
> 
> Lastly, I think showing popular and new games in a single row would look nicer. Four games for each category would be enough for this. Maybe we can implement swiping if we have time. What do you think?

I'll start working on property of the day as well. Adding newly launched games frontend wise will also not be a problem  since I've already implemented the lists component but I think sliding might take up more time. We can create a low priority issue for it and do it after more important functionalities are implemented. I've followed the mockups but I'll add your reccomandations for the page formatting as well after I complete the game page

**Comment by asyasusen** [comment]
> @asyasusen how about naming main-page branch front-end/main-page or something like this to be more clear, since later we can be confused by names in backend-frontend?

Done

**Comment by sonerkuyar** [comment]
I dont know it's only designing but page has no -nearly zero- functionality. Could you at least bind search bar to any state?

**Comment by sonerkuyar** [comment]
I'm joining this issue due to @asyasusen didnt't respond me. I already edited some parts of the page I'll continue.

**Comment by asyasusen** [comment]
> I'm joining this issue due to @asyasusen didnt't respond me. I already edited some parts of the page I'll continue.

So are you doing the search funtionality or should I do it?

**Comment by sonerkuyar** [comment]
I dealed with also backend structure of  this structure backend structure and frontend navigations is complete but page was not completed so I postponed this feature to next milestone.Btw, Its a bit problematic for me to change the code when its only depend on hard-coded assets. For the next milestone Even if we're desiginin we should consider the next step by looking at the class diagrams of backend wheras their backend functions are not complete.

**Comment by asyasusen** [comment]
I've completed adding endpoints for game of the day, new games and popular games. I've also implemented the core structure of game page and I'm awaiting Isil to finish implementing for that endpoint. We don't have the review end-points yet I'm waiting on them to implement the friend's reviews and recent review's part. 

Also we should disscuss the more-games part I think it might crash to load thousands of games and we can just scratch that since we already have search functionality (I'll remind this topic during the meeting)

**Comment by asyasusen** [comment]
Everything is completed on my end

**Comment by ahmetfirat23** [comment]
lgtm

**Comment by sonerkuyar** [comment]
I've made lots of changes and fixes an hour before the deadline you can see all of them at this PR. 
[UI developments](https://github.com/bounswe/bounswe2024group12/pull/242)
